src/FieldDefinition.hack

```diff
@@ -34,4 +35,16 @@ final class FieldDefinition<TParent, TRet, TResolved> implements IFieldDefinitio
 
         return await $this->type->resolveAsync($value, $field, $vars);
     }
+
+    public function getName(): string {
+        return $this->name;
+    }
+
+    public function getType(): Introspection\__Type {
+        if ($this->type is Types\NullableInputType<_> || $this->type is Types\NullableOutputType<_, _>) {
+            return $this->type;
+        }
+
+        return new Introspection\NonNullable($this->type);
```

Comment:
We only need to output info for non nulls during introspection. With this we don't impact normal execution by wrapping classes unnecessarily with `NonNullable`, we just do it during introspection.